teamNewDev/jobhunt
# Setup Prettier Pre commit

#### What does this PR do?
This PR adds a pre-commit hook that runs prettier against code to be committed. This would ensure the enforcement of our code convention.
This can re-format your files that are marked as “staged” via git add before you commit.

#### Description of Task to be Completed
- install prettier-eslint, prettier-eslint-cli, and lint-staged as a dev dependency
- install husky with `npx husky-init`. This command will install Husky as a development dependency and create a simple pre-commit hook in .husky/pre-commit that you can edit.
- Inside `.husky/pre-commit` replace `npm test` with `npx lint-staged`.
- Add `"prepare": "husky install"` inside the script object in `package.json`
- In the root directory of your project, create the file `.lintstagedrc.json` with the following contents:
```
{
  "*.js": [
    "prettier --write",
    "eslint"
  ],
  "*.html": [
    "eslint",
    "prettier --write"
  ],
  "*.css": "prettier --write"
}
```
- Add a lint-staged script in `package.json` with the following content:
```
 "*.js": [
      "npm run lint",
      "prettier --write",
      "git add ."
    ]
```

#### How should this be manually tested?
1. Clone the repository.
2. Switch to the `pre-commit` branch
3. Upon running `npm install` in the command line, all project dependencies should be installed in the local dev environment
4. After changes in a js file, upon `git add .` and `git commit`, pre-commit should run and report any formatting errors or warnings.

#### Any background context you want to provide?
Pre-commit hooks can be used to ensure that code adheres to specific formatting and style guidelines. This can include enforcing indentation, spacing, line breaks, and naming conventions. By using pre-commit hooks for code formatting and standards, developers can ensure that code is consistent and easy to read.

#### What are the relevant GitHub Issues?
#6 

#### Screenshots (if appropriate)
![image](https://github.com/teamNewDev/jobhunt/assets/40843759/27d983cd-7d28-4574-baf7-1d9d761a9d12)

#### Error Testing
According the rule set in the eslint file, semi colon should not be allowed. Then try git commit with semi-colon in a file then pre-commit throws a warning.
Example:
![image](https://github.com/teamNewDev/jobhunt/assets/40843759/9a81ad11-8d9d-49c1-a4d0-926535e9d61a)
